ethcore/src/blockchain/blockchain.rs

```diff
@@ -351,11 +333,13 @@ impl BlockProvider for BlockChain {
 
 	/// Returns numbers of blocks containing given bloom.
 	fn blocks_with_bloom(&self, bloom: &Bloom, from_block: BlockNumber, to_block: BlockNumber) -> Vec<BlockNumber> {
```

Comment:
Imho would be best to return hashes here, we convert to numbers, then in `client` we pass `numbers` to `blockchain::logs` method that converts to hashes again. Seems pretty wasteful